python/doc/source/operators.rst

```diff
@@ -1,53 +1,61 @@
 Operators
 =========
 
-An ERDOS operator receives data on :py:class:`ReadStreams <erdos.ReadStream>`,
-and sends processed data on :py:class:`WriteStreams <erdos.WriteStream>`.
+ERDOS operators process received data, and use streams to broadcast Messages 
+to downstream operators.
 We provide a standard library of operators for common dataflow patterns
-under :py:mod:`erdos.operators`.
+under :py:mod:`erdos.operator`.
 While the standard operators are general and versatile, some applications may
 implement custom operators to better optimize performance and take
 fine-grained control over exection.
 
-All operators must inherit from the :py:class:`~erdos.Operator` base class and
-implement :py:meth:`~erdos.Operator.__init__` and
-:py:meth:`~erdos.Operator.connect` methods.
-
-* :py:meth:`~erdos.Operator.__init__` takes all
-  :py:class:`ReadStreams <erdos.ReadStream>` from which the operator receives
-  data, all :py:class:`WriteStreams <erdos.WriteStream>` on which the operator
-  sends data, and any other arguments passed when calling
-  :py:meth:`~erdos.Operator.connect`.
-  Within :py:meth:`~erdos.Operator.__init__`, the state should be initialized,
-  and callbacks may be registered across
-  :py:class:`ReadStreams <erdos.ReadStream>`.
-
-* The :py:meth:`~erdos.Operator.connect` method takes
-  :py:class:`ReadStreams <erdos.ReadStream>` and returns
-  :py:class:`WriteStreams <erdos.WriteStream>`
-  which are all later passed to :py:meth:`~erdos.Operator.__init__` by ERDOS.
-  The :py:class:`ReadStreams <erdos.ReadStream>` and
-  :py:class:`WriteStreams <erdos.WriteStream>`
-  must appear in the same order as in :py:meth:`~erdos.Operator.__init__`.
+Operators are structures which implement a certain communication pattern. For example, 
```

Comment:
Could you add that operators are implemented as classes, and must subclass the built-in operator type that specifies its communication pattern?

Also, describe the different types of communication patterns (Source, Sink, OneInOneOut, etc) in an unordered list